src/test/java/com/marklogic/kafka/connect/sink/SendWriteFailureRecordsToDlqKafkaTest.java

```diff
@@ -0,0 +1,111 @@
+package com.marklogic.kafka.connect.sink;
+
+import com.marklogic.client.query.DeleteQueryDefinition;
+import com.marklogic.junit5.spring.SimpleTestConfig;
+import kafka.server.KafkaConfig$;
+import net.mguenther.kafka.junit.EmbeddedKafkaCluster;
+import net.mguenther.kafka.junit.KeyValue;
+import net.mguenther.kafka.junit.ReadKeyValues;
+import net.mguenther.kafka.junit.SendKeyValues;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static net.mguenther.kafka.junit.EmbeddedConnectConfig.kafkaConnect;
+import static net.mguenther.kafka.junit.EmbeddedKafkaCluster.provisionWith;
+import static net.mguenther.kafka.junit.EmbeddedKafkaClusterConfig.newClusterConfig;
+import static net.mguenther.kafka.junit.ObserveKeyValues.on;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SendWriteFailureRecordsToDlqKafkaTest extends AbstractIntegrationSinkTest {
+
+    // Declared by AbstractSpringMarkLogicTest
+    @Autowired
+    private SimpleTestConfig testConfig;
+
+    private final String ML_COLLECTION = "kafka-data";
+    private final String TOPIC = "test-topic";
+    private final String DLQ_TOPIC = "test-dlq";
+    private final String KEY = String.format("key-%s", UUID.randomUUID());
+    private final Integer NUM_RECORDS = 1;
+
+    private EmbeddedKafkaCluster kafka;
+
+    @BeforeEach
+    void setupKafka() {
+        provisionKafkaWithConnectAndMarkLogicConnector();
+        kafka.start();
+        cleanUpDataForTest();
+    }
+
+    @AfterEach
+    void tearDownKafka() {
+        kafka.stop();
+    }
+
+    @Test
+    void failedBatchesShouldGoToTheDlq() throws InterruptedException {
+        sendSomeJsonMessages(NUM_RECORDS);
```

Comment:
I think it'd be good for this test to send a batch containing one valid document and one invalid document. And then we verify that neither was written to MarkLogic, and the one Throwable from the failed batch ended up in the DLQ... which makes me wonder, should the URIs of that failed batch going into the message that goes to the DLQ, like as part of the headers? That would provide very good visibility as to what failed.